api/users/permissions.py

```diff
@@ -0,0 +1,15 @@
+from website.models import OAuth2App
+from rest_framework import permissions
+
+
+class OwnerOnly(permissions.BasePermission):
```

Comment:
I'd support consolidating permissions in general; having a ton of scattered, highly specific implementations seems like a code smell. (especially if those rules are implemented differently across different API versions or views)

Along those lines, let me know if there are any changes I can make to the OAuth2 app model to support a more centralized and consistent set of permissions controls. Thanks.